### PR TITLE
gomplate: init at 3.5.0

### PR DESCRIPTION
**Pull request by jlesquembre**
###### Motivation for this change
Add [gomplate](https://docs.gomplate.ca)

I was able to build it locally. I also did some tests and the binary seems to work fine, but I don't have much experience with go packages, feedback about the nix expression would be appreciated. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jlesquembre** [comment]
I updated the PR to use `fetchFromGithub` instead of `fetchgit`, to make it more deterministic

File: pkgs/development/tools/gomplate/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
```
Comment (doronbehar) [inline pkgs/development/tools/gomplate/default.nix:3]: You should be able to switch to `buildGoModule` as that's the recommended approach for projects that support it.

File: pkgs/development/tools/gomplate/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  pname = "gomplate";
+  version = "3.5.0";
```
Comment (doronbehar) [inline pkgs/development/tools/gomplate/default.nix:5]: 3.6.0 is available https://github.com/hairyhenderson/gomplate/releases

File: pkgs/development/tools/gomplate/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  preBuild = ''
+    export buildFlagsArray+=(
+      "-ldflags=
+        -w -s
+        -X ${goPackagePath}/version.Version=${version}
+        -X ${goPackagePath}/version.GitCommit=${rev}")
+  '';
+
+  goPackagePath = "github.com/hairyhenderson/gomplate";
```
Comment (doronbehar) [inline pkgs/development/tools/gomplate/default.nix:26]: With `buildGoModule` that's not needed.

File: pkgs/development/tools/gomplate/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  subPackages = [ "cmd/gomplate" ];
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    description = "A flexible commandline tool for template rendering. Supports lots of local and remote datasources";
+    homepage = https://gomplate.ca/;
```
Comment (doronbehar) [inline pkgs/development/tools/gomplate/default.nix:32]:
```suggestion
    homepage = "https://gomplate.ca/";
```

https://github.com/NixOS/rfcs/pull/45

File: pkgs/development/tools/gomplate/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+     rev = "v${version}";
+     sha256 = "0q3fssgwn7p95pn4nh5qajssrc55jqf5gwq6cpirfim0c339pl95";
+   };
+
+  preBuild = ''
+    export buildFlagsArray+=(
+      "-ldflags=
+        -w -s
+        -X ${goPackagePath}/version.Version=${version}
+        -X ${goPackagePath}/version.GitCommit=${rev}")
```
Comment (doronbehar) [inline pkgs/development/tools/gomplate/default.nix:23]: If you really want the executable to have the git commit defined in it, wouldn't it be almost as good to use:

```suggestion
        -X ${goPackagePath}/version.GitCommit=${version}")
```
?

**Review by doronbehar — CHANGES_REQUESTED** [review]

**Comment by jlesquembre** [comment]
Thanks for your review @doronbehar . I created a new derivation for 3.6 version, but never created a PR for it, doing it now. I'm closing this, we can continue on #85157

**Pull request closed without merging**